hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -626,14 +626,12 @@ public void terminate(String reason, Exception cause, boolean clearMetrics, bool
       Threads.shutdown(initThread, this.sleepForRetries);
     }
     Collection<ReplicationSourceShipper> workers = workerThreads.values();
+
     for (ReplicationSourceShipper worker : workers) {
       worker.stopWorker();
-      if(worker.entryReader != null) {
+      if (worker.entryReader != null) {
         worker.entryReader.setReaderRunning(false);
       }
-    }
-
-    for (ReplicationSourceShipper worker : workers) {
```

Comment:
Semantics of this have changed, but I'm not seeing conversation that indicates that it was intentional.

Before: we would stop all workers, then wait for them all to be stopped. Each worker could stop itself concurrently. Now, for each worker, we request a stop and then wait for it to be stopped, then move on to the next worker.

I don't _think_ this is a big deal, but wanted to call it out.

Reply:
Sorry for not explaining this before. Yeah, just thought there was not real gains on doing two iterations, as at the end, we would need to make sure all workers were indeed stopped. I can revert it back, if you perfer.

Reply:
Nope, all good.